src/domains/services/validation.service.ts

```diff
@@ -283,11 +284,70 @@ export class ValidationService extends BaseService {
         },
       );
 
-      const status = this.determineDiagnosticsStatus(diagnostics, options);
+      // Additional diagnostics for Avro payload schemas using avsc
+      const extraDiagnostics: Diagnostic[] = [];
+      try {
+        if (typeof document?.allMessages === 'function') {
+          const allMessages = document.allMessages();
+          for (const message of allMessages.values()) {
+            const payload = message.payload();
+            if (!payload) {continue;}
+
+            const schemaFormat = message.schemaFormat();
+            if (!schemaFormat) {continue;}
+
+            const isAvro = (/application\/vnd\.apache\.avro\+(json|yaml)/i).test(schemaFormat);
```

Comment:
This string matches only if a `+json` or `+yaml` is present.
As I understand the spec, a valid Avro schema format can also be `application/vnd.apache.avro;version=1.9.0`

see: https://www.asyncapi.com/docs/reference/specification/v3.0.0#multiFormatSchemaFormatTable